app/core/SanitizationMiddleware.ts

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+import { JsonRpcMiddleware, JsonRpcRequest } from 'json-rpc-engine';
+import { addHexPrefix } from 'ethereumjs-util';
+
+interface TransactionParamsLike {
+  from: string;
+  to: string;
+  value: string;
+  data: string;
+  gas: string;
+  gasPrice: string;
+  nonce: string;
+  fromBlock: string;
+  toBlock: string;
+  address: string;
+  topics: string[];
+  [k: string]: string | string[];
+}
+
+// we use this to clean any custom params from the txParams
+const permittedKeys = [
+  'from',
+  'to',
+  'value',
+  'data',
+  'gas',
+  'gasPrice',
+  'nonce',
+  'fromBlock',
+  'toBlock',
+  'address',
+  'topics',
+];
+
+function cloneTxParams(txParams: TransactionParamsLike | any) {
```

Comment:
Nit: Any type union including `any` can be reduced to `any`.

Instead maybe we can use the type `Record<PropertyKey, unknown>` here. By the time this function is called, we have confirmed that `txParams` is a non-null object, which is basically what `Record<PropertyKey, unknown>` would be.

Type checking doesn't really apply here because the `JsonRpcRequest` type we're using here has `any` in it (I wouldn't recommend bothering to try and improve that in this PR; I tried once by introducing `@metamask/utils` but it caused build failures on Android). But at least that type reflects what we ought to know by this point.

```suggestion
function cloneTxParams(txParams: Record<PropertyKey, unknown>) {
```